### PR TITLE
Removed old physics material library code

### PR DESCRIPTION
**Pull request by moraaar**
**This PR is part of the work to refactoring physics materials. It's NOT merging to development branch.**

Rename new material classes now that old code has been removed.

Signed-off-by: moraaar <moraaar@amazon.com>

### CODE REVIEW

File: Gems/PhysX/Code/Tests/ShapeColliderComponentTests.cpp [hunk 1]
```diff
@@ -518,8 +518,7 @@ namespace PhysXEditorTests
         return CreateActiveGameEntityFromEditorEntity(rigidBodyEditorEntity.get());
     }
 
-    // LYN-1241 - Test disabled due to AZ_Error reports about MaterialLibrary being not found in the AssetCatalog
-    TEST_F(PhysXEditorFixture, DISABLED_EditorShapeColliderComponent_PolygonPrismForceRegion_AppliesForceAtRuntime)
+    TEST_F(PhysXEditorFixture, EditorShapeColliderComponent_PolygonPrismForceRegion_AppliesForceAtRuntime)
```
Comment (moraaar) [inline Gems/PhysX/Code/Tests/ShapeColliderComponentTests.cpp:521]: All unit tests passed, including this re-enabled one.

File: Code/Framework/AzFramework/AzFramework/Physics/Configuration/SystemConfiguration.h [hunk 2]
```diff
@@ -56,5 +53,11 @@ namespace AzPhysics
         // helpers for edit context
         AZ::u32 OnMaxTimeStepChanged();
         float GetFixedTimeStepMax() const;
+
+        // Padding with a 16 byte aligned type to make the class aligned to 16 bytes too.
+        // This is used instead of adding alignas(16) to the struct because that generates
+        // warnings everywhere the struct is used indicating that padding is added, which 
+        // causes compilation errors.
+        AZ::Vector4 m_padding;
```
Comment (hultonha) [inline Code/Framework/AzFramework/AzFramework/Physics/Configuration/SystemConfiguration.h:61]: uber nit: Consider renaming this to `m_unusedPadding` to be clear this is purely there for padding reasons and not for anything else

**Review by hultonha — APPROVED** [review]
LGTM :+1:

File: Code/Framework/AzFramework/AzFramework/Physics/Configuration/SystemConfiguration.h (hunk 2 above)
Comment (moraaar) [reply Code/Framework/AzFramework/AzFramework/Physics/Configuration/SystemConfiguration.h:61]: will rename.

**Review by SergeyAMZN — APPROVED** [review]

**Pull request merged**